music-journal/components/AddMemory.js

```diff
@@ -12,20 +12,23 @@ const AddMemory = ({ handleSubmit, text, setText, song, setSong }) => {
                     <Row className="justify-content-center">
                         <Form.Group controlId="formSong">
                             <Form.Label>Write a song name</Form.Label>
-                            <Form.Control 
-                                as="textarea" 
-                                onChange={({target}) => setSong(target.value)} 
-                                rows="1" 
+                            <Form.Control
+                                as="textarea"
+                                onChange={({ target }) => {
+                                    setSong(target.value)
+                                    handleAutocomplete()
```

Comment:
can this be done in "useEffect"? it only seems to be being called on load in the parent component, I guess a child useEffect? idk

Reply:
You could do something like this in the parent component. You can add more than one useEffect to a component.
```
useEffect(() => {
 if (song > 2) {
   const response = await searchTracks(song)
   setSearchResults(response)
}
}, [song])
```

Reply:
This would run each time the value of "song" changes.

Reply:
Ahhh now I properly understand the empty array passed to the other useEffect! This doesn't seem to work though :/

TypeError: r is not a function
    at ro (framework.baa41d4dbf5d52db897c.js:1)
    at yu (framework.baa41d4dbf5d52db897c.js:1)
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at vu (framework.baa41d4dbf5d52db897c.js:1)
    at eu (framework.baa41d4dbf5d52db897c.js:1)
    at framework.baa41d4dbf5d52db897c.js:1
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at Kl (framework.baa41d4dbf5d52db897c.js:1)
eo @ framework.baa41d4dbf5d52db897c.js:1
i.componentDidCatch.n.callback @ framework.baa41d4dbf5d52db897c.js:1
di @ framework.baa41d4dbf5d52db897c.js:1
io @ framework.baa41d4dbf5d52db897c.js:1
mu @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
pu @ framework.baa41d4dbf5d52db897c.js:1
eu @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
Kl @ framework.baa41d4dbf5d52db897c.js:1
Hl @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
D @ framework.baa41d4dbf5d52db897c.js:1
V @ framework.baa41d4dbf5d52db897c.js:1
W @ framework.baa41d4dbf5d52db897c.js:1
Zt @ framework.baa41d4dbf5d52db897c.js:1
Jt @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
F @ framework.baa41d4dbf5d52db897c.js:1
Xt @ framework.baa41d4dbf5d52db897c.js:1
main-77542faf49fb381c51cc.js:1 TypeError: r is not a function
    at ro (framework.baa41d4dbf5d52db897c.js:1)
    at yu (framework.baa41d4dbf5d52db897c.js:1)
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at vu (framework.baa41d4dbf5d52db897c.js:1)
    at eu (framework.baa41d4dbf5d52db897c.js:1)
    at framework.baa41d4dbf5d52db897c.js:1
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at Kl (framework.baa41d4dbf5d52db897c.js:1)
oe @ main-77542faf49fb381c51cc.js:1
fn @ main-77542faf49fb381c51cc.js:1
value @ main-77542faf49fb381c51cc.js:1
i.componentDidCatch.n.callback @ framework.baa41d4dbf5d52db897c.js:1
di @ framework.baa41d4dbf5d52db897c.js:1
io @ framework.baa41d4dbf5d52db897c.js:1
mu @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
pu @ framework.baa41d4dbf5d52db897c.js:1
eu @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
Kl @ framework.baa41d4dbf5d52db897c.js:1
Hl @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
D @ framework.baa41d4dbf5d52db897c.js:1
V @ framework.baa41d4dbf5d52db897c.js:1
W @ framework.baa41d4dbf5d52db897c.js:1
Zt @ framework.baa41d4dbf5d52db897c.js:1
Jt @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
F @ framework.baa41d4dbf5d52db897c.js:1
Xt @ framework.baa41d4dbf5d52db897c.js:1
framework.baa41d4dbf5d52db897c.js:1 TypeError: n is not a function
    at framework.baa41d4dbf5d52db897c.js:1
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at ao (framework.baa41d4dbf5d52db897c.js:1)
    at so (framework.baa41d4dbf5d52db897c.js:1)
    at mu (framework.baa41d4dbf5d52db897c.js:1)
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at pu (framework.baa41d4dbf5d52db897c.js:1)
    at eu (framework.baa41d4dbf5d52db897c.js:1)
eo @ framework.baa41d4dbf5d52db897c.js:1
i.componentDidCatch.n.callback @ framework.baa41d4dbf5d52db897c.js:1
di @ framework.baa41d4dbf5d52db897c.js:1
io @ framework.baa41d4dbf5d52db897c.js:1
mu @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
pu @ framework.baa41d4dbf5d52db897c.js:1
eu @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
Kl @ framework.baa41d4dbf5d52db897c.js:1
Hl @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
D @ framework.baa41d4dbf5d52db897c.js:1
V @ framework.baa41d4dbf5d52db897c.js:1
W @ framework.baa41d4dbf5d52db897c.js:1
Zt @ framework.baa41d4dbf5d52db897c.js:1
Jt @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
F @ framework.baa41d4dbf5d52db897c.js:1
Xt @ framework.baa41d4dbf5d52db897c.js:1
main-77542faf49fb381c51cc.js:1 TypeError: n is not a function
    at framework.baa41d4dbf5d52db897c.js:1
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at ao (framework.baa41d4dbf5d52db897c.js:1)
    at so (framework.baa41d4dbf5d52db897c.js:1)
    at mu (framework.baa41d4dbf5d52db897c.js:1)
    at t.unstable_runWithPriority (framework.baa41d4dbf5d52db897c.js:1)
    at jl (framework.baa41d4dbf5d52db897c.js:1)
    at pu (framework.baa41d4dbf5d52db897c.js:1)
    at eu (framework.baa41d4dbf5d52db897c.js:1)
oe @ main-77542faf49fb381c51cc.js:1
fn @ main-77542faf49fb381c51cc.js:1
value @ main-77542faf49fb381c51cc.js:1
i.componentDidCatch.n.callback @ framework.baa41d4dbf5d52db897c.js:1
di @ framework.baa41d4dbf5d52db897c.js:1
io @ framework.baa41d4dbf5d52db897c.js:1
mu @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
pu @ framework.baa41d4dbf5d52db897c.js:1
eu @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
Kl @ framework.baa41d4dbf5d52db897c.js:1
Hl @ framework.baa41d4dbf5d52db897c.js:1
(anonymous) @ framework.baa41d4dbf5d52db897c.js:1
D @ framework.baa41d4dbf5d52db897c.js:1
V @ framework.baa41d4dbf5d52db897c.js:1
W @ framework.baa41d4dbf5d52db897c.js:1
Zt @ framework.baa41d4dbf5d52db897c.js:1
Jt @ framework.baa41d4dbf5d52db897c.js:1
t.unstable_runWithPriority @ framework.baa41d4dbf5d52db897c.js:1
jl @ framework.baa41d4dbf5d52db897c.js:1
F @ framework.baa41d4dbf5d52db897c.js:1
Xt @ framework.baa41d4dbf5d52db897c.js:1
framework.baa41d4dbf5d52db897c.js:1 Uncaught TypeError: r is not a function
    at onChange (memory-31b264fb7d747c2fa361.js:1)
    at Object.o (framework.baa41d4dbf5d52db897c.js:1)
    at p (framework.baa41d4dbf5d52db897c.js:1)
    at framework.baa41d4dbf5d52db897c.js:1
    at y (framework.baa41d4dbf5d52db897c.js:1)
    at ot (framework.baa41d4dbf5d52db897c.js:1)
    at Array.forEach (<anonymous>)
    at it (framework.baa41d4dbf5d52db897c.js:1)
    at ut (framework.baa41d4dbf5d52db897c.js:1)
    at mt (framework.baa41d4dbf5d52db897c.js:1)